store/src/commonMain/kotlin/com/dropbox/external/store5/concurrent/Lightswitch.kt

```diff
@@ -0,0 +1,26 @@
+package com.dropbox.external.store5.concurrent
+
+import kotlinx.coroutines.sync.Semaphore
+
+internal class Lightswitch {
+    private var counter = 0
+    private val mutex = Semaphore(1)
```

Comment:
why not mutex= Mutex? I think its confusing to init a mutex as a sephamore.

Reply:
Replaced all semaphores with mutex